src/Microsoft.Data.SqlClient/tests/ManualTests/ProviderAgnostic/MultipleResultsTest/MultipleResultsTest.cs

```diff
@@ -2,366 +2,154 @@
-                filter[1] = filter[0].Replace("mscorlib", "System");
-                filter[2] = filter[0].Replace("mscorlib", "System.Data");
-                filter[3] = filter[0].Replace("mscorlib", "System.Data.OracleClient");
-                filter[4] = filter[0].Replace("mscorlib", "System.Xml");
-                _outputFilter = filter;
-            }
-
-            for (int i = 0; i < filter.Length; ++i)
-            {
-                builder.Replace(filter[i], "");
-            }
+            using SqlConnection connection = new SqlConnection((new SqlConnectionStringBuilder(DataTestUtility.TCPConnectionString) { MultipleActiveResultSets = true }).ConnectionString);
+            using SqlCommand command = connection.CreateCommand();
+            ConcurrentQueue<string> messages = new ConcurrentQueue<string>();
+
+            connection.InfoMessage += (object sender, SqlInfoMessageEventArgs args) =>
+                messages.Enqueue(args.Message);
+
+            connection.Open();
+
+            command.CommandText = s_sqlStatement;
+
+            // SqlDataReader will drain every result set it's told to, throwing exceptions and printing messages as it goes.
+            using SqlDataReader reader = command.ExecuteReader();
+
+            // Result set 1: a result set and a message of '0'.
+            // Result set 2: a message of '1' and an exception.
+            AdvanceReader(reader, messages, ResultSet1_Message, ResultSet2_Message, ResultSet2_Error, finalBlock: false);
+
+            // Result set 3: a result set and a message of '3'.
+            // Result set 4: a message of '4' and an exception.
+            AdvanceReader(reader, messages, ResultSet3_Message, ResultSet4_Message, ResultSet4_Error, finalBlock: false);
+
+            // Result set 5: a result set and a message of '5'.
+            // Result set 6: a message of '6' and an exception.
+            AdvanceReader(reader, messages, ResultSet5_Message, ResultSet6_Message, ResultSet6_Error, finalBlock: false);
+
+            // Result set 7: a result set and a message of '7'.
+            // Result set 8: a message of '8' and an exception.
+            AdvanceReader(reader, messages, ResultSet7_Message, ResultSet8_Message, ResultSet8_Error, finalBlock: false);
+
+            // Result set 9: a result set and a message of '9'.
+            // Result set 10: a message of '10' and an exception.
+            AdvanceReader(reader, messages, ResultSet9_Message, ResultSet10_Message, ResultSet10_Error, finalBlock: true);
+
+            // One message following the final result set
+            Assert.True(messages.TryDequeue(out string lastMessage));
+            Assert.Empty(messages);
+            Assert.Equal(ResultSet11_Message, lastMessage);
         }
 
-        /// <summary>
-        ///  special wrapper for the text writer to replace single "\n" with "\n"
-        /// </summary>
-        private sealed class CarriageReturnLineFeedReplacer : TextWriter
```

Comment:
HOLY SMOKES what the heck is this 🗑️ ?????? 😂

Reply:
In the original test, I think someone was trying to account for Linux/Windows differences by replacing CRLF with LF. It's odd though - and unnecessary, since SqlException.CreateException uses `Environment.NewLine` anyway.